### ISSUE TITLE
PHP Fatal Error -Uncaught PDOException when catching 404's

### ISSUE THREAD

**Issue by SkermBE** [barrelstrength/sprout#309]
### Description

Whenever we enable the `Log 404 Redirects` we get the following error when landing on a 404 page:

```
PHP Fatal Error – [yii\base\ErrorException](https://www.yiiframework.com/doc-2.0/yii-base-errorexception.html)
Uncaught PDOException: SQLSTATE[23000]: Integrity constraint violation: 1452 Cannot add or update a child row: a foreign key constraint fails (`db`.`structureelements`, CONSTRAINT `fk_qfzbxcufjisgijodwqnsyktqjfzbxpxbevuc` FOREIGN KEY (`structureId`) REFERENCES `structures` (`id`) ON DELETE CASCADE) in /var/www/html/vendor/yiisoft/yii2/db/Command.php:1302
```

### How to reproduce

1. Install the plugin
2. Enable the `Log 404 Redirects` from the settings page
3. Browse to any URL thad does not exist, triggering a 404
4. See the error apear

### Sprout Version

4.46.1 (lite)

### Craft Version

4.7.3

### Database Type Version

MySQL

**Comment by SkermBE** [comment]
It seems we have the same issue when creating a new redirect:

```
Cannot add or update a child row: a foreign key constraint fails (`database_name`.`structureelements`, CONSTRAINT `fk_qfzbxcufjisgijodwqnsyktqjfzbxpxbevuc` FOREIGN KEY (`structureId`) REFERENCES `structures` (`id`) ON DELETE CASCADE)
)
```

**Comment by BenParizek** [comment]
@SkermBE I'm not sure where this was introduced but the Redirect Elements are stored as a Structure in the Craft DB and it appears the UID in your project config is no longer referencing the Structure that Redirects expect.

I'll need to ask you to troubleshoot this on your end, but I've just released Redirects v4.46.2 and added two console commands that can help get this resolved.

If this is a fresh install and you don't have any redirects in your database the following console command will create a new Structure and add that new UID to your config:

```
./craft sprout-module-redirects/support/hard-reset-structure-uid
```

If you have redirects in your db that you want to keep, the following console command will _try_ to match the expected ID and update the config.

```
./craft sprout-module-redirects/support/refresh-structure-uid
```

In both cases, the updated UID will be added in the project config. Let me know if you can't get this issue resolved with the above.

**Comment by BenParizek** [comment]
Closing this out but please open a new ticket or comment if the issue continues.
